crates/arrow_sql_gen/src/postgres.rs

```diff
@@ -235,8 +251,95 @@ fn map_column_type_to_array_builder(column_type: &Type) -> Box<dyn ArrayBuilder>
         Type::FLOAT8 => Box::new(arrow::array::Float64Builder::new()),
         Type::TEXT => Box::new(arrow::array::StringBuilder::new()),
         Type::BOOL => Box::new(arrow::array::BooleanBuilder::new()),
-        Type::NUMERIC => Box::new(arrow::array::Decimal128Builder::new()),
+        // TODO: Figure out how to handle decimal scale and precision, it isn't specified as a type in postgres types
+        Type::NUMERIC => Box::new(
+            arrow::array::Decimal128Builder::new()
+                .with_precision_and_scale(38, 9)
+                .unwrap_or_default(),
+        ),
         Type::TIMESTAMP => Box::new(arrow::array::TimestampMicrosecondBuilder::new()),
         _ => unimplemented!("Unsupported column type {:?}", column_type),
     }
 }
+
+struct BigDecimalFromSql {
+    inner: BigDecimal,
+    scale: u16,
+}
+
+impl BigDecimalFromSql {
+    fn to_decimal_128(&self) -> Option<i128> {
+        (&self.inner * 10i128.pow(u32::from(self.scale))).to_i128()
+    }
+}
+
+impl<'a> FromSql<'a> for BigDecimalFromSql {
```

Comment:
Was this code taken from somewhere? I would prefer to rely on a library that does this for us rather than having it in-house. Or if we do keep it, we should add unit tests to cover it.

Reply:
Nope, I had to dig in and learn about how Postgres represents numeric values myself. I searched pretty far and wide and rust_decimal implements `FromSql` but it only supports up to 96 bits, not 128 which is what we need for Decimal128.

The only implementation I could find of `FromSql` for BigDecimal is in Diesel, but to use it we'd have to use Diesel as an ORM throughout our code which would require knowing schemas at compile time, which we don't: https://docs.rs/diesel/latest/diesel/

Reply:
Added 2 basic tests below, need more to cover edge cases